linux-mainline: Update base image and fix build failures

alistair23 opened the pull request.

alistair23 committed:
linux-mainline: Update base image and fix build failures

Signed-off-by: Alistair Francis <alistair@alistair23.me>

Eeems commented:
```
[   DEBUG] toltec.builder: build(): In file included from /opt/x-tools/arm-remarkable-linux-gnueabihf/arm-remarkable-linux-gnueabihf/sysroot/usr/include/features.h:489,
[   DEBUG] toltec.builder: build(): from /opt/x-tools/arm-remarkable-linux-gnueabihf/arm-remarkable-linux-gnueabihf/sysroot/usr/include/bits/libc-header-start.h:33,
[   DEBUG] toltec.builder: build(): from /opt/x-tools/arm-remarkable-linux-gnueabihf/arm-remarkable-linux-gnueabihf/sysroot/usr/include/stdio.h:27,
[   DEBUG] toltec.builder: build(): from certs/extract-cert.c:15:
[   DEBUG] toltec.builder: build(): /opt/x-tools/arm-remarkable-linux-gnueabihf/arm-remarkable-linux-gnueabihf/sysroot/usr/include/gnu/stubs.h:7:11: fatal error: gnu/stubs-soft.h: No such file or directory
[   DEBUG] toltec.builder: build(): 7 | # include <gnu/stubs-soft.h>
[   DEBUG] toltec.builder: build(): |           ^~~~~~~~~~~~~~~~~~
[   DEBUG] toltec.builder: build(): compilation terminated.
```

Eeems commented:
This might be due to trying to use the system pkgconfig instead of the toolchain version.

alistair23 commented:
Fixed! (Hopefully)

alistair23 commented:
Anything else for me to do here?

Eeems commented:
Just the standard testing of the package build in the pipeline on 2.15 and 3.3.2 is required. I don't have capacity right now to do any testing.

Eeems commented:
So one issue on 3.3.2 is that this kernel doesn't provide uinput, which breaks compatibility with the current version of oxide as well as keyd (#744).

@alistair23 you have two options here to resolve this:

1. Update the kernel to include uinput
2. Mark this package as conflicting with oxide, and we update #744 to indicate it conflicts with linux-mainline

Eeems commented:
@alistair23 since this hasn't been updated, I'm going to move this to draft until you pick it back up.

alistair23 committed:
linux-mainline: Enable INPUT_UINPUT

Enable INPUT_UINPUT as requested [1].

1: https://github.com/toltec-dev/toltec/pull/904#issuecomment-2336752065

Signed-off-by: Alistair Francis <alistair@alistair23.me>

alistair23 commented:
I just added uinput support

Eeems commented:
@alistair23 it looks like you undid the version update when you added uinput?

alistair23 commented:
Fixed!

Eeems committed:
Merge branch 'testing' into alistair/build-fixup

Eeems committed:
Merge branch 'testing' into alistair/build-fixup

Eeems committed:
Merge branch 'testing' into alistair/build-fixup

Eeems committed:
Merge branch 'testing' into alistair/build-fixup
